drivers/sdhc/sdhci_common.c

```diff
@@ -474,6 +545,45 @@ static void sdhci_prepare_data(struct sdhci_common *sdhci_ctx, struct sdhc_comma
 	sys_write16(transfer_mode, sdhci_ctx->reg_base + SDHCI_TRANSFER_MODE);
 }
 
+static int sdhci_dma_check(struct sdhci_common *sdhci_ctx, struct sdhc_data *data)
+{
+	uint32_t trans_bytes;
+
+	if (!sdhci_ctx->f_use_dma) {
+		return 0;
+	}
+
+#if defined(CONFIG_64BIT)
+	/* Only 32bit DMA supported */
+	if (((uint64_t)data->data >> 32)) {
```

Comment:
Is it possible that we have a 32-bit virtual address and a 64-bit DMA address for it?
Shouldn't you check the DMA address here instead of the virtual one?

Reply:
fixed